src/WCSimPrimaryGeneratorAction.cc

```diff
@@ -29,681 +29,725 @@ vector<string> tokenize( string separators, string input );
 
 inline vector<string> readInLine(fstream& inFile, int lineSize, char* inBuf)
 {
-  // Read in line break it up into tokens
-  inFile.getline(inBuf,lineSize);
-  return tokenize(" $\r", inBuf);
+    // Read in line break it up into tokens
+    // Any line starting with # is ignored
+
+    while(true)
+    {
+    	if(inFile.getline(inBuf,lineSize))
+    	{    
+    	   // G4cout<<inBuf<<G4endl;
+           if(inBuf[0]!='#')
+              return tokenize(" $\r", inBuf);
+      }
+      else
+      {
+       G4cout<<" error reading from file "<<G4endl;
+       vector<string> nullLine;
+       return nullLine;
+   }
+}
 }
 
 inline float atof( const string& s ) {return std::atof( s.c_str() );}
 inline int   atoi( const string& s ) {return std::atoi( s.c_str() );}
 
 WCSimPrimaryGeneratorAction::WCSimPrimaryGeneratorAction(
-					  WCSimDetectorConstruction* myDC)
-  :myDetector(myDC), vectorFileName("")
+    WCSimDetectorConstruction* myDC)
```

Comment:
There are also many changes here in WCSimPrimaryGeneratorAction that are just spacing (although here I also see added code). Could this be cleaned up a bit?